crp_APL/ctrl_vehicle_control/src/ctrl_vehicle_control.cpp

```diff
@@ -0,0 +1,151 @@
+   
+    }
+
+    void ego_vehicle_callback(const crp_msgs::msg::Ego input_msg)
+    {
+        input.vxEgo = input_msg.twist.twist.linear.x;
+        input.egoSteeringAngle = input_msg.road_wheel_angle_front;
+
+        input.egoPoseGlobal[0] = input_msg.pose.pose.position.x;
+        input.egoPoseGlobal[1] = input_msg.pose.pose.position.y;
+        double quaternion[4];
+        quaternion[0] = input_msg.pose.pose.orientation.x; 
+        quaternion[1] = input_msg.pose.pose.orientation.y; 
+        quaternion[2] = input_msg.pose.pose.orientation.z; 
+        quaternion[3] = input_msg.pose.pose.orientation.w; 
+        double theta = m_geometricOperator.transformQuatToEuler(quaternion);
+        input.egoPoseGlobal[2] = theta;
+    }
+
+    void loop()
+    {
+        // parameter assignments
+        params.ffGainOffsetGround = this->get_parameter("/ctrl/ffGainOffsetGround").as_double();
+        params.ffGainSlope = this->get_parameter("/ctrl/ffGainSlope").as_double();
+        params.ffLookAheadTime = this->get_parameter("/ctrl/ffLookAheadTime").as_double();
+        params.ffMinLookAheadDistance = this->get_parameter("/ctrl/ffMinLookAheadDistance").as_double();
+        params.steeringAngleLPFilter = this->get_parameter("/ctrl/steeringAngleLPFilter").as_double();
+        params.fbLookAheadTime = this->get_parameter("/ctrl/fbLookAheadTime").as_double();
+        params.fbPGain = this->get_parameter("/ctrl/fbPGain").as_double();
+        params.fbDGain = this->get_parameter("/ctrl/fbDGain").as_double();
+        params.fbIGain = this->get_parameter("/ctrl/fbIGain").as_double();
+        params.fbThetaGain = this->get_parameter("/ctrl/fbThetaGain").as_double();
+        params.fbMinLookAheadDistance = this->get_parameter("/ctrl/fbMinLookAheadDistance").as_double();
+        params.fbIntegralLimit = this->get_parameter("/ctrl/fbIntegralLimit").as_double();
+        params.trajectory_distance = this->get_parameter("/ctrl/trajectory_distance").as_double();
+        params.debugKPIs = this->get_parameter("/ctrl/debugKPIs").as_bool();
+
+        m_compensatoryModel.run(input, output, params);
+
+        // steering angle and steering angle gradiant
+        ctrl_cmd.lateral.steering_tire_angle = output.steeringAngleTarget;
+        ctrl_cmd.lateral.steering_tire_rotation_rate = 0.0f;
+        ctrl_cmd.longitudinal.speed = input.target_speed;
+
+        cmd_pub->publish(ctrl_cmd);
+    }
+    rclcpp::TimerBase::SharedPtr timer_;
+    rclcpp::Publisher<autoware_auto_control_msgs::msg::AckermannControlCommand>::SharedPtr cmd_pub;
+    
+    rclcpp::Subscription<autoware_auto_planning_msgs::msg::Trajectory>::SharedPtr traj_sub;
+    rclcpp::Subscription<nav_msgs::msg::Odometry>::SharedPtr odom_sub;
+    rclcpp::Subscription<crp_msgs::msg::Ego>::SharedPtr ego_vehicle_sub;
+    rclcpp::Subscription<autoware_auto_control_msgs::msg::AckermannControlCommand>::SharedPtr cmd_dummy_sub;
```

Comment:
Missing m_ prefix and camelCase with member variables
```suggestion
    rclcpp::Subscription<autoware_auto_control_msgs::msg::AckermannControlCommand>::SharedPtr m_sub_cmdDummy;
```